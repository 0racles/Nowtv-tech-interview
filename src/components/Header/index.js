import React from 'react';
import styled from 'styled-components';
import logo from '../../../logo.png';

const Wrapper = styled.div`
	background-color: #222;
	color: white;
	height: 150px;
	padding: 20px;
`;

const StyledImage = styled.img`
	max-height: 80px;
  max-width: 80%;
`;

const Header = () => {
  return (
    <Wrapper className="header-title">
      <StyledImage src={logo} className="header-logo" alt="logo" />
      <h2>Interview Assessment</h2>
    </Wrapper>
  );
};

export default Header;
