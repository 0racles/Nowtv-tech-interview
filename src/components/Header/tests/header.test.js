import Header from '../index';

describe('Header', () => {
  it('should match snapshot', done => {
    setTimeout(done, 4000);
    const renderedItem = shallow(<Header />);
    expect(renderedItem).toMatchSnapshot();
  });
});
