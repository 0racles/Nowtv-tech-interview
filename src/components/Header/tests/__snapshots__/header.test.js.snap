exports[`Header should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Unknown
          alt="logo"
          className="header-logo"
          src="logo.png" />,
        <h2>
          Interview Assessment
</h2>,
      ],
      "className": "header-title",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "logo",
          "className": "header-logo",
          "src": "logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
	max-height: 80px;
  max-width: 80%;
",
            ],
          },
          "displayName": "styled.img",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "img",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Interview Assessment",
        },
        "ref": null,
        "rendered": "Interview Assessment",
        "type": "h2",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
	background-color: #222;
	color: white;
	height: 150px;
	padding: 20px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Unknown
            alt="logo"
            className="header-logo"
            src="logo.png" />,
          <h2>
            Interview Assessment
</h2>,
        ],
        "className": "header-title",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "logo",
            "className": "header-logo",
            "src": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
	max-height: 80px;
  max-width: 80%;
",
              ],
            },
            "displayName": "styled.img",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "img",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Interview Assessment",
          },
          "ref": null,
          "rendered": "Interview Assessment",
          "type": "h2",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
	background-color: #222;
	color: white;
	height: 150px;
	padding: 20px;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
